### PR TITLE
runtime: add warp-time feature for "fast" runtime

### PR DESCRIPTION
**Pull request by mnaamani**
Added a "warp-time" cargo feature for use with production runtime configuration to address https://github.com/Joystream/joystream/issues/4894

To test build and run local chain (with production configration)

```sh
WASM_BUILD_TOOLCHAIN=nightly-2022-11-15 cargo build --release --features warp-time
```

```
./target/release/joystream-node --chain prod-test --tmp --alice --rpc-cors=all --pruning=archive
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1204404358291458/1205678974517544) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by freakstatic — COMMENTED** [review]
@mnaamani are you sure you want to have "two names" for the same "feature"?
warp-time vs fast-block-production

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -226,12 +226,26 @@ runtime-benchmarks = [
...
-# Staging and testing configurations
-# configuration suitable for staging networks and playground
+# Configuration suitable for staging networks and playground
 staging-runtime = []
 playground-runtime = []
-# configuration suitable for integration testing
-testing-runtime = []
+# Configuration suitable for automated integration testing only
+testing-runtime = [
+    "fast-block-production"
```
Comment (freakstatic) [inline runtime/Cargo.toml:234]: will not this make testing-runtime with fast-block-production by default?

**Comment by freakstatic** [comment]
oh maybe I misunderstood, so fast-block-production it's the regular testing environment and warp-time is the one with block production of 1s right?

**Review by freakstatic — APPROVED** [review]

File: runtime/Cargo.toml (hunk 1 above)
Comment (mnaamani) [reply runtime/Cargo.toml:234]: That is already the case. The testing runtime uses 1s blocks (but doesn't warp the timing)

**Comment by mnaamani** [comment]
> oh maybe I misunderstood, so fast-block-production it's the regular testing environment and warp-time is the one with block production of 1s right?

~Correct.~
Not exactly.

`testing-runtime` feature and `warp-time` both enable 1s block times
testing-runtime uses different runtime constants for council/election/proposal periods than production.
warp-time if enabled with production runtime will use the production runtime constants and for periods which are defined using hours!,minutes!,days! macros will not be real minutes,hours,days .. they will be warped times.. so for example `hours!(1)` will be number of blocks (based on 6s) so 60 * 60 / 6 =  600 blocks. But since a blocks are actually at 1s intervals, that is actually 600 / 60 = 10 (real) minutes.

**Pull request merged**